File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Deque<Integer> stack = new ArrayDeque<>();

        for (String token : tokens) {
            if (token.equals("+")) {
                stack.push(stack.pop() + stack.pop());
            } else if (token.equals("-")) {
                int last = stack.pop();
                stack.push(stack.pop() - last);
            } else if (token.equals("*")) {
                stack.push(stack.pop() * stack.pop());
            } else if (token.equals("/")) {
                int last = stack.pop();
                stack.push(stack.pop() / last);
            } else {
                stack.push(Integer.parseInt(token));
            }
        }

        return stack.pop();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.kt
class Solution {
    fun evalRPN(tokens: Array<String>): Int {
        val stack = ArrayDeque<Int>()

        for (token in tokens) {
            when (token) {
                "+" -> stack.addLast(stack.removeLast() + stack.removeLast())
                "-" -> {
                    val last = stack.removeLast()
                    stack.addLast(stack.removeLast() - last)
                }
                "*" -> stack.addLast(stack.removeLast() * stack.removeLast())
                "/" -> {
                    val last = stack.removeLast()
                    stack.addLast(stack.removeLast() / last)
                }
                else -> stack.addLast(token.toInt())
            }
        }

        return stack.last()
    }
}